switch_model/generators/extensions/hydro_simple.py

```diff
@@ -88,6 +90,34 @@ def define_components(mod):
     mod.have_minimal_hydro_params = BuildCheck(
         mod.HYDRO_GEN_TS,
         rule=lambda m, g, ts: (g,ts) in m.HYDRO_GEN_TS_RAW)
+    # Generate a warning if the input files specify timeseries for renewable
+    # plant capacity factors that extend beyond the expected lifetime of the
+    # plant. This could be caused by simple logic to build input files, or
+    # could indicate that the user expects those plants to operate longer
+    # than indicated.
+    def _warn_on_extra_HYDRO_GEN_TS(m):
+        extra_indexes = set(m.HYDRO_GEN_TS_RAW) - set(m.HYDRO_GEN_TS)
+        num_impacted_generators = len(set([g for g,t in extra_indexes]))
+        extraneous = {g: [] for (g,t) in extra_indexes}
+        for (g,t) in extra_indexes:
+            extraneous[g].append(t)
+        pprint = "\n".join(
+            "* {}: {}".format(g, tps) for g, tps in extraneous.items())
+        warning_msg = (
+            "{} hydro plants have data specified "
+            "in timeseries after they are slated for retirement. This "
+            "could indicate a benign issue where the process that built "
+            "the dataset used simplified logic and/or didn't know the "
+            "scheduled retirement date. If you expect those datapoints to "
+            "be useful, then those plants need longer lifetimes (or "
+            "options to build new capacity when the old capacity reaches "
+            "the provided end-of-life date).\n".format(num_impacted_generators))
+        if extra_indexes:
+            logging.warning(warning_msg)
+            logging.info("Plants with extra timepoints:\n{}".format(pprint))
+        return(True)
+    mod.warn_on_extra_HYDRO_GEN_TS = BuildCheck(
+        rule=_warn_on_extra_HYDRO_GEN_TS)
```

Comment:
In https://github.com/switch-model/switch/pull/115/commits/728938566b4552f61366eb4292a3448c6d16658a#r341685397 I recommended using `warning` as our default logging level, and defining the `info` level as being equivalent to our current `--verbose` flag. For the case shown here--where we allow data for extra timepoints but it's not considered good practice for purists--I would show the messages at the `debug` level, not the `info` or `warning` level. If we've decided that this behavior is OK, then we shouldn't hassle the user with warning messages every time they run the model.